lang/src/org/partiql/lang/eval/Thunk.kt

```diff
@@ -526,24 +549,7 @@ internal class LegacyThunkFactory(
         try {
             block()
         } catch (e: EvaluationException) {
-            when {
-                e.errorContext == null ->
-                    throw EvaluationException(
-                        message = e.message,
-                        errorCode = e.errorCode,
-                        errorContext = errorContextFrom(sourceLocation),
-                        cause = e,
-                        internal = e.internal
-                    )
-                else -> {
-                    // Only add source location data to the error context if it doesn't already exist
-                    // in [errorContext].
-                    if (!e.errorContext.hasProperty(Property.LINE_NUMBER)) {
-                        sourceLocation?.let { fillErrorContext(e.errorContext, sourceLocation) }
-                    }
-                    throw e
-                }
-            }
+            throw populateErrorContext(e, sourceLocation)
```

Comment:
Corrects a small bug I discovered where line/column information was not being added to some `EvaluationException`s.